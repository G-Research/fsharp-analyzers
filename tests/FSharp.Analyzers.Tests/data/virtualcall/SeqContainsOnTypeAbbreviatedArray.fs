module M

[<RequireQualifiedAccess>]
type SyntaxNode =
    | SynPat of int
    | SynType of int

type SyntaxVisitorPath = SyntaxNode array

let f (path: SyntaxVisitorPath) =
    Seq.length path // should warn
