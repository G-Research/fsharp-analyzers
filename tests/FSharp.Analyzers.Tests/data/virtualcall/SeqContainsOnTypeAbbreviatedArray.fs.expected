GRA-VIRTUALCALL-001 | Warning | (11,4 - 11,19) | Consider replacing the call of Seq.length with a function from the Array module to avoid the costs of virtual calls. | [(11,4 - 11,7) Array]
